tests/unit/test_schema_registry.py

```diff
@@ -0,0 +1,257 @@
+                return self.parsed_schema3
+            raise ValueError(f"Unexpected object {schema}")
+
+        self.schema_registry.resolve_and_parse = MagicMock(side_effect=resolve_and_parse_mock)
+
+    def check_schema_compatibility(
+        self,
+        new_schema: ValidatedTypedSchema,
+        subject: Subject,
+    ) -> SchemaCompatibilityResult:
+        return self.schema_registry.check_schema_compatibility(new_schema, subject)
+
+
+async def test_schema_compatible_in_transitive_mode() -> None:
+    # Given
+    schema_registry = MockedSchemaRegistry(CompatibilityModes.FULL_TRANSITIVE)
+    new_schema = ValidatedTypedSchema.parse(SchemaType.JSONSCHEMA, '{"type": "array"}')
+    subject = Subject("subject")
+
+    # Don't test the actual compatibility checks
+    result = Mock(spec=SchemaCompatibilityResult)
+    result.compatibility = SchemaCompatibilityType.compatible
+    SchemaCompatibility.check_compatibility = Mock(return_value=result)
+
+    # When
+    schema_registry.check_schema_compatibility(new_schema, subject)
+
+    # Then
+    assert result.compatibility is SchemaCompatibilityType.compatible
+
+    # All 3 schemas are checked against
+    SchemaCompatibility.check_compatibility.assert_has_calls(
+        [
+            mock.call(
+                old_schema=schema_registry.parsed_schema1,
+                new_schema=new_schema,
+                compatibility_mode=schema_registry.compatibility_mode,
+            ),
+            mock.call(
+                old_schema=schema_registry.parsed_schema2,
+                new_schema=new_schema,
+                compatibility_mode=schema_registry.compatibility_mode,
+            ),
+            mock.call(
+                old_schema=schema_registry.parsed_schema3,
+                new_schema=new_schema,
+                compatibility_mode=schema_registry.compatibility_mode,
+            ),
+        ]
+    )
+
+
+async def test_schema_incompatible_in_transitive_mode() -> None:
+    # Given
+    schema_registry = MockedSchemaRegistry(CompatibilityModes.FULL_TRANSITIVE)
+    new_schema = ValidatedTypedSchema.parse(SchemaType.JSONSCHEMA, '{"type": "array"}')
+    subject = Subject("subject")
+
+    # Don't test the actual compatibility checks
+    result = Mock(spec=SchemaCompatibilityResult)
+    result.compatibility = SchemaCompatibilityType.incompatible
+    SchemaCompatibility.check_compatibility = Mock(return_value=result)
+
+    # When
+    schema_registry.check_schema_compatibility(new_schema, subject)
+
+    # Then
+    assert result.compatibility is SchemaCompatibilityType.incompatible
+
+    # Only one schema is checked against (first fail stops all checks)
+    SchemaCompatibility.check_compatibility.assert_has_calls(
+        [
+            mock.call(
+                old_schema=schema_registry.parsed_schema1,
+                new_schema=new_schema,
+                compatibility_mode=schema_registry.compatibility_mode,
+            ),
+        ]
+    )
```

Comment:
kinda the same comment, I would like to see two test. 1 where the schemas are compatible and the api reply green, one where the schemas are 2 by 2 compatible but not fully transitive compatible and the api reply red

Reply:
Moved to `tests/integration/test_schema_compatibility.py`